src/main/java/mate/academy/bookshop/service/impl/BookServiceImpl.java

```diff
@@ -19,12 +23,19 @@
 @RequiredArgsConstructor
 public class BookServiceImpl implements BookService {
     private final BookSpecificationBuilder specificationBuilder;
+    private final CategoryRepository categoryRepository;
     private final BookRepository bookRepository;
     private final BookMapper bookMapper;
 
     @Override
     public BookDto save(CreateBookRequestDto requestDto) {
         Book book = bookMapper.toModel(requestDto);
+        Set<Category> categories = new HashSet<>();
+        for (Long categoryId : requestDto.getCategories()) {
+            categories.add(categoryRepository.findById(categoryId).orElseThrow(()
+                    -> new EntityNotFoundException("There are no category with id " + categoryId)));
+        }
```

Comment:
Better to give DB check if requested categories are existed, instead of get all of them from DB.

Reply:
Agree